File: backend/src/main/java/com/yagubogu/member/dto/MemberProfileResponse.java
package com.yagubogu.member.dto;

import com.yagubogu.member.domain.Member;
import java.time.LocalDate;

public record MemberProfileResponse(
        String nickname,
        LocalDate enterDate,
        String profileImageUrl,
        String favoriteTeam,
        MemberRepresentativeBadgeResponse representativeBadge,
        MemberVictoryFairyResponse victoryFairy,
        MemberCheckInResponse checkIn
) {
    public static MemberProfileResponse from(
            final Member member,
            final MemberRepresentativeBadgeResponse badge,
            final MemberVictoryFairyResponse victoryFairy,
            final MemberCheckInResponse checkIn) {
        return new MemberProfileResponse(
                member.getNickname().getValue(),
                member.getCreatedAt().toLocalDate(),
                member.getImageUrl(),
                member.getTeam().getShortName(),
                badge,
                victoryFairy,
                checkIn
        );
    }
}

File: android/app/src/main/java/com/yagubogu/data/dto/response/member/MemberProfileResponse.kt
package com.yagubogu.data.dto.response.member

import kotlinx.datetime.LocalDate
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class MemberProfileResponse(
    @SerialName("nickname")
    val nickname: String, // 회원 닉네임
    @SerialName("enterDate")
    val enterDate: LocalDate, // 회원 가입일 (YYYY-MM-DD)
    @SerialName("profileImageUrl")
    val profileImageUrl: String, // 프로필 이미지 주소
    @SerialName("favoriteTeam")
    val favoriteTeam: String, // 응원하는 팀
    @SerialName("representativeBadge")
    val representativeBadge: ProfileRepresentativeBadgeDto?, // 대표 배지 정보
    @SerialName("victoryFairy")
    val victoryFairyProfile: VictoryFairyProfileDto, // 승리 요정 정보
    @SerialName("checkIn")
    val checkIn: MemberCheckInDto, // 직관 인증 정보
)
